src/main/java/uk/gov/companieshouse/exemptions/delta/upsert/UpsertClient.java

```diff
@@ -35,20 +36,10 @@ void upsert(UpsertRequest request) {
             client.privateDeltaCompanyAppointmentResourceHandler()
                     .upsertCompanyExemptionsResource(request.getPath(), request.getBody())
                     .execute();
-        } catch (ApiErrorResponseException e) {
-            if(e.getStatusCode() / 100 == 5) {
-                logger.error(String.format("Server error returned with status code: [%s] when upserting delta", e.getStatusCode()));
-                throw new RetryableException("Server error returned when upserting delta", e);
-            } else {
-                logger.error(String.format("Upsert client error returned with status code: [%s] when upserting delta", e.getStatusCode()));
-                throw new NonRetryableException("UpsertClient error returned when upserting delta", e);
-            }
-        } catch (IllegalArgumentException e) {
-            logger.error("Illegal argument exception caught when handling API response");
-            throw new RetryableException("Server error returned when upserting delta", e);
+        }  catch (ApiErrorResponseException e) {
+            handler.handle(e);
```

Comment:
These two exception catch blocks can we unified using `||`

Reply:
The ResponseHandler class handles these two exceptions separately.
To do this would require modification to the handle method (add conditional check on what type of exception it is).

Not sure if its worth that amount of effort to get rid of one line.